File: q2.c
#include<stdio.h>

int main(){

int n;
scanf("%d",&n);

printf("ENTER VALUES OF 1ST VECTOR: ");
float vec1[n],vec2[n];

for (int i=0;i<n;i++){
	scanf("%f",&vec1[i]);}

 printf("ENTER VALUES OF 2ND VECTOR: ");
for (int i=0;i<n;i++){
        scanf("%f",&vec2[i]);}

float dp = 0;

for (int i=0;i<n;i++){
	dp += (vec1[i])*(vec2[i]);}
printf("%f is the dot product of the two vectors.",dp);

return 0;
}

File: q2.py
"""
Write a program to accept a positive integer n, two vectors of length
n into two arrays, and find their dot product. Example run:
Enter the value of n: 5
Enter the values of the first vector: 3 -1 10 2.5 6
Enter the values of the second vector: 0.5 4 1 0 -0.5
The dot product is 4.5.
"""
import numpy as np

vec1 = np.loadtxt("vec1.dat")
vec2 = np.loadtxt("vec2.dat")

dot_product = np.dot(vec1, vec2)

print(f"The dot product is {dot_product}.")
